src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIMarsConnection.cs

```diff
@@ -303,30 +321,37 @@ public void HandleReceiveComplete(SNIPacket packet, uint sniErrorCode)
                         if (_currentHeader.flags == (byte)SNISMUXFlags.SMUX_FIN)
                         {
                             _sessions.Remove(_currentHeader.sessionId);
+                            SqlClientEventSource.Log.TrySNITraceEvent("SNIMarsConnection.HandleReceiveComplete | SNI | FIN | MARS Session Id {0}, SMUX_FIN flag received, Current Header Session Id {1} removed", _lowerHandle?.ConnectionId, _currentHeader?.sessionId);
                         }
                         else
                         {
                             currentSession = _sessions[_currentHeader.sessionId];
+                            SqlClientEventSource.Log.TrySNITraceEvent("SNIMarsConnection.HandleReceiveComplete | SNI | INFO | MARS Session Id {0}, Current Session assigned to Session Id {1}", _lowerHandle?.ConnectionId, _currentHeader?.sessionId);
                         }
                     }
 
                     if (currentHeader.flags == (byte)SNISMUXFlags.SMUX_DATA)
                     {
                         currentSession.HandleReceiveComplete(currentPacket, currentHeader);
+                        SqlClientEventSource.Log.TrySNITraceEvent("SNIMarsConnection.HandleReceiveComplete | SNI | INFO | SMUX_DATA | MARS Session Id {0}, Current Session {1} completed receiving Data", _lowerHandle?.ConnectionId, _currentHeader?.sessionId);
                     }
 
                     if (_currentHeader.flags == (byte)SNISMUXFlags.SMUX_ACK)
                     {
                         try
                         {
                             currentSession.HandleAck(currentHeader.highwater);
+                            SqlClientEventSource.Log.TrySNITraceEvent("SNIMarsConnection.HandleReceiveComplete | SNI | INFO | SMUX_ACK | MARS Session Id {0}, Current Session {1} handled ack", _lowerHandle?.ConnectionId, _currentHeader?.sessionId);
                         }
                         catch (Exception e)
                         {
+                            SqlClientEventSource.Log.TrySNITraceEvent("SNIMarsConnection.HandleReceiveComplete | SNI | ERR | MARS Session Id {0}, Exception occurred: {2}", _currentHeader?.sessionId, e?.Message);
```

Comment:
`e` is never null I think inside a catch block.

Reply:
Just for safety and consistency I added it here, it doesn't impact driver anyways.